src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4664,7 +4664,9 @@ internal enum AddressKind
                 return false;
             }
 
-            if (!field.IsReadOnly)
+            Debug.Assert(field.RefKind is RefKind.None or RefKind.RefReadOnly);
+
+            if (field.RefKind == RefKind.None && !field.IsReadOnly)
```

Comment:
>field.RefKind == RefKind.None [](http://example.com/codeflow?start=16&length=29)

It looks like at this point we are always supposed to return `false` for ```RefKind.RefReadOnly```. Consider doing that explicitly and reverting changes below.